pkg/operator/operator.go

```diff
@@ -213,3 +219,41 @@ func getVnetGUID(cfg *auth.Config, subnetID string) (string, error) {
 	}
 	return *vnet.Properties.ResourceGUID, nil
 }
+
+// WaitForCRDs waits for the required CRDs to be available with a timeout
+func WaitForCRDs(ctx context.Context, client *discovery.DiscoveryClient, timeout time.Duration, log logr.Logger) error {
+	var requiredCRDs = []schema.GroupVersionResource{
+		{Group: "karpenter.sh", Version: "v1", Resource: "nodepools"},
+		{Group: "karpenter.sh", Version: "v1", Resource: "nodeclaims"},
+		{Group: "karpenter.azure.com", Version: "v1alpha2", Resource: "aksnodeclasses"},
+	}
+
+	log.WithValues("timeout", timeout).Info("waiting for required CRDs to be available")
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	for _, crd := range requiredCRDs {
+		crdName := fmt.Sprintf("%s.%s/%s", crd.Resource, crd.GroupVersion().Group, crd.Version)
+		if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
minor, style: maybe just assign `err` here and check it on the subsequent line to avoid this:
`}); err != nil {`

not a big deal though.

Reply:
Fixed